src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -74,16 +74,28 @@ public UInt256 Hash
             sizeof(long) +  //NetworkFee
             sizeof(uint);   //ValidUntilBlock
 
-        public int Size => HeaderSize +
-            Attributes.GetVarSize() +   //Attributes
-            Cosigners.GetVarSize() +    //Cosigners
-            Script.GetVarSize() +       //Script
-            Witnesses.GetVarSize();     //Witnesses
+        /// <summary>
+        /// Assigned when tx is deserialized
+        /// </summary>
+        private int _size;
+        public int Size
+        {
+            get
+            {
+                return _size == 0 ? HeaderSize +
+                  Attributes.GetVarSize() +   //Attributes
+                  Cosigners.GetVarSize() +    //Cosigners
+                  Script.GetVarSize() +       //Script
+                  Witnesses.GetVarSize()      //Witnesses
+                  : _size;
+            }
+        }
 
         void ISerializable.Deserialize(BinaryReader reader)
         {
             DeserializeUnsigned(reader);
             Witnesses = reader.ReadSerializableArray<Witness>();
+            _size = (int)reader.BaseStream.Position;
```

Comment:
It is incorrect. Because we may be deserializing an array element.

Reply:
How about this way?

```csharp
uint startPosition =  (int)reader.BaseStream.Position
DeserializeUnsigned(reader);
Witnesses = reader.ReadSerializableArray<Witness>();
_size = (int)reader.BaseStream.Position  - startPosition;
```